pageserver/src/pgdatadir_mapping.rs

```diff
@@ -714,7 +714,12 @@ impl Timeline {
         lsn: Lsn,
         ctx: &RequestContext,
     ) -> Result<HashMap<String, Bytes>, PageReconstructError> {
-        match self.get_switch_aux_file_policy() {
+        let current_policy = AuxFilePolicy::int_to_opt(
+            self.last_aux_file_policy
+                .load(std::sync::atomic::Ordering::SeqCst),
+        )
+        .unwrap_or(AuxFilePolicy::V1);
```

Comment:
I was opposing the fact that we now have spread around:

1. `load(seqcst)`
2. `int_to_opt` + `unwrap_or`

Instead we should have a single type which gives you that end result (`AuxFilePolicy`).

Are the `int_to_opt` because you don't want to use `impl TryFrom<usize> for AuxFilePolicy` or why? The `Option::unwrap_or` would be `Result::unwrap_or`.

Perhaps the helper type needs to contain the atomic, like `RuntimeAuxFilesPolicy` or something like that.

Reply:
Yeah I think it's a good idea to split SwitchAuxFilePolicy and RuntimeAuxFilePolicy. I did this int_to_opt hack because they are not two separate types for now, and `None` means different things in these two settings.